jupyterlab_automl/jupyterlab_automl/service.py

```diff
@@ -1,284 +1,247 @@
-                    {
-                        ChartInfo.name.value: Months(month).name,
-                        ChartInfo.amount.value: amount,
-                    }
-                )
-            for day, amount in dict(
-                column_spec.data_stats.timestamp_stats.granular_stats[
-                    "day_of_week"
-                ].buckets
-            ).items():
-                day_chart.append(
-                    {
-                        ChartInfo.name.value: Days(day).name,
-                        ChartInfo.amount.value: amount,
-                    }
-                )
-            for hour, amount in dict(
-                column_spec.data_stats.timestamp_stats.granular_stats[
-                    "hour_of_day"
-                ].buckets
-            ).items():
-                time_chart.append(
-                    {
-                        ChartInfo.name.value: str(hour) + ":00",
-                        ChartInfo.amount.value: amount,
-                    }
-                )
-            return [month_chart, day_chart, time_chart]
+  """Provides an authenticated AutoML Client and project info"""
+
+  _instance = None
+
+  def __init__(self):
+    client_options = {"api_endpoint": API_ENDPOINT}
+    self._dataset_client = aiplatform_v1alpha1.DatasetServiceClient(
+        client_options=client_options)
+    self._model_client = aiplatform_v1alpha1.ModelServiceClient(
+        client_options=client_options)
+    self._pipeline_client = aiplatform_v1alpha1.PipelineServiceClient(
+        client_options=client_options)
+    self._gcs_client = storage.Client()
+
+    self._project = AuthProvider.get().project
+
+    self._parent = "projects/{}/locations/us-central1".format(self._project)
+    self._gcs_bucket = "jupyterlab-ucaip-data-{}".format(
+        hashlib.md5(self._project.encode('utf-8')).hexdigest())
+
+  @property
+  def dataset_client(self):
+    return self._dataset_client
+
+  @property
+  def model_client(self):
+    return self._model_client
+
+  @classmethod
+  def get(cls):
+    if not cls._instance:
+      cls._instance = AutoMLService()
+    return cls._instance
+
+  def get_models(self):
+    models = []
+    for model in self._model_client.list_models(parent=self._parent).models:
+      model_type = 'OTHER'
+      if 'image' in model.metadata_schema_uri:
+        model_type = 'IMAGE'
+      elif 'tables' in model.metadata_schema_uri:
+        model_type = 'TABLE'
+      models.append({
+            "id": model.name,
+            "displayName": model.display_name,
+            "pipelineId": model.training_pipeline,
+            "createTime": (model.create_time.year, model.create_time.month, model.create_time.day, model.create_time.hour, model.create_time.minute, model.create_time.second),
+            "updateTime": (model.update_time.year, model.update_time.month, model.update_time.day, model.update_time.hour, model.update_time.minute, model.update_time.second),
```

Comment:
changes times to timestamps to serialize and create Date objects in the TypeScript